File: backend/src/main/java/com/ll/hotel/domain/member/admin/service/AdminHotelService.java
package com.ll.hotel.domain.member.admin.service;

import com.ll.hotel.domain.hotel.hotel.entity.Hotel;
import com.ll.hotel.domain.hotel.hotel.repository.HotelRepository;
import com.ll.hotel.domain.member.admin.dto.request.AdminHotelRequest;
import com.ll.hotel.global.exceptions.ErrorCode;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class AdminHotelService {
    private final HotelRepository hotelRepository;

    public Page<Hotel> findAllPaged(int page) {
        Pageable pageable = PageRequest.of(page, 10);

        Page<Hotel> pagedHotel = hotelRepository.findAll(pageable);

        if (!pagedHotel.hasContent()) {
            ErrorCode.PAGE_NOT_FOUND.throwServiceException();
        }

        return pagedHotel;
    }

    public Hotel findById(Long id) {
        return hotelRepository.findById(id)
                .orElseThrow(ErrorCode.HOTEL_NOT_FOUND::throwServiceException);
    }

    @Transactional
    public void approve(Hotel hotel, AdminHotelRequest adminhotelRequest) {
        hotel.setHotelStatus(adminhotelRequest.hotelStatus());
    }

    public void flush() {
        hotelRepository.flush();
    }
}

File: backend/src/main/java/com/ll/hotel/domain/member/admin/service/AdminHotelService.kt
package com.ll.hotel.domain.member.admin.service

import com.ll.hotel.domain.hotel.hotel.repository.HotelRepository
import com.ll.hotel.domain.member.admin.dto.request.AdminHotelRequest
import com.ll.hotel.domain.member.admin.dto.response.AdminHotelResponse
import com.ll.hotel.global.exceptions.ErrorCode
import com.ll.hotel.standard.page.dto.PageDto
import jakarta.transaction.Transactional
import org.springframework.data.domain.PageRequest
import org.springframework.stereotype.Service

@Service
class AdminHotelService(
    private val hotelRepository: HotelRepository
) {
    fun findAllPaged(page: Int): PageDto<AdminHotelResponse.Summary> {
        val pageable = PageRequest.of(page, 10)
        val pagedHotel = hotelRepository.findAll(pageable)

        if (pagedHotel.hasContent().not()) {
            ErrorCode.PAGE_NOT_FOUND.throwServiceException()
        }

        val pagedHotelSummaries = pagedHotel.map(AdminHotelResponse.Summary::from)

        return PageDto(pagedHotelSummaries)
    }

    fun findById(id: Long): AdminHotelResponse.Detail {
        val hotel = hotelRepository.findById(id)
            .orElseThrow(ErrorCode.HOTEL_NOT_FOUND::throwServiceException)

        return AdminHotelResponse.Detail.from(hotel)
    }

    @Transactional
    fun approve(id: Long, adminHotelRequest: AdminHotelRequest
    ): AdminHotelResponse.ApprovalResult {
        val hotel = hotelRepository.findById(id)
            .orElseThrow(ErrorCode.HOTEL_NOT_FOUND::throwServiceException)

        hotel.hotelStatus = adminHotelRequest.hotelStatus
        hotelRepository.flush()

        return AdminHotelResponse.ApprovalResult.from(hotel)
    }
}